bindings_node/src/encoded_content.rs

```diff
@@ -0,0 +1,81 @@
+use napi::bindgen_prelude::Uint8Array;
+use napi_derive::napi;
+use std::{collections::HashMap, ops::Deref};
+use xmtp_proto::xmtp::mls::message_contents::{ContentTypeId, EncodedContent};
+
+#[derive(Clone)]
+#[napi(object)]
+pub struct NapiContentTypeId {
+  pub authority_id: String,
+  pub type_id: String,
+  pub version_major: u32,
+  pub version_minor: u32,
+}
+
+impl From<ContentTypeId> for NapiContentTypeId {
+  fn from(content_type_id: ContentTypeId) -> NapiContentTypeId {
+    NapiContentTypeId {
+      authority_id: content_type_id.authority_id,
+      type_id: content_type_id.type_id,
+      version_major: content_type_id.version_major,
+      version_minor: content_type_id.version_minor,
+    }
+  }
+}
+
+impl Into<ContentTypeId> for NapiContentTypeId {
+  fn into(self) -> ContentTypeId {
+    ContentTypeId {
+      authority_id: self.authority_id,
+      type_id: self.type_id,
+      version_major: self.version_major,
+      version_minor: self.version_minor,
+    }
+  }
+}
+
+#[derive(Clone)]
+#[napi(object)]
+pub struct NapiEncodedContent {
+  pub r#type: Option<NapiContentTypeId>,
+  pub parameters: HashMap<String, String>,
+  pub fallback: Option<String>,
+  pub compression: Option<i32>,
+  pub content: Uint8Array,
+}
+
+impl From<EncodedContent> for NapiEncodedContent {
+  fn from(content: EncodedContent) -> NapiEncodedContent {
+    let r#type = match content.r#type {
+      Some(v) => Some(v.into()),
+      None => None,
+    };
+
+    NapiEncodedContent {
+      r#type,
+      parameters: content.parameters,
+      fallback: content.fallback,
+      compression: content.compression,
+      content: content.content.into(),
+    }
+  }
+}
+
+impl Into<EncodedContent> for NapiEncodedContent {
```

Comment:
same question as above, but perhaps this is different since i'm converting a into a `Vec<u8>`?

Reply:
All From's automatically generated the inverse `Into`. One of those neat magic Rust features.